docs/docs-guides/deployment-workflow.md

```diff
@@ -0,0 +1,35 @@
+Our documentation is hosted on [GitHub Pages](https://pages.github.com/). Whenever a new push to **gh-pages** branch happens, it will deploy a new version of the documentation. 
+
+The older versions of the documentation will still remain in the **gh-pages** branch, because we are supporting the docs versioning by [mike](https://github.com/jimporter/mike). That's also the reason, why we recommend making any changes and pushes to **gh-pages** branch with the usage of this tool. 
```

Comment:
Is this still true?
Do we really need a gh-pages branch?
Can't we just deploy a new "release" of the docs whenever there's a new release of claudie itself?

Reply:
This is still true.
When we run `mike deploy <version> --push`, the tool itself push to **gh-pages** and the new version of the docs is deployed, because the GitHub Pages monitors **gh-pages** branch for new releases.
The **gh-pages** branch is also required, because the mike tool provides docs versioning only through this branch.

Reply:
Ok, but at least we don't have to know that, and we can simply rely on the automation in the pipeline, right ?
if that is the case, can you mention in this doc that it's all automated in the release pipeline?

Reply:
Sure, I have changed this part of the doc in the last commit.